### DISCUSSION TITLE
How to write text right to left for arabic

### DISCUSSION THREAD

**Discussion by alikzalikz** [markdoc/markdoc#325] [Q&A] [answered]
hello
thanks for this useful tools
i want to create a blog with Arabic language
that's need write text Right-to-left
in HTML can use `<p dir="rtl"><p/>` tag for this reason
but in pure markdown i cant
what i do?

**Comment by rpaul-stripe** [comment, marked as answer]
You can override Markdoc's built-in `paragraph` node type and add a `dir` attribute:

```
const config = {
  nodes: {
    paragraph: {
      ...Markdoc.nodes.paragraph,
      attributes: {
        dir: {type: "string"}
      }
    }
  }
};

const doc = Markdoc.parse(example);
const output = Markdoc.transform(doc, config);
```

This will make it so that you can use an annotation in a paragraph to set the value of `dir`:

```
This is a paragraph in a Markdown document with a direction annotation. {% dir="rtl" %}
```

If you want to make it apply `dir="rtl"` by default to every paragraph, you can set that as the default value of the attribute in the schema, and then you don't have to add the annotation in the Markdown:

```
const config = {
  nodes: {
    paragraph: {
      ...Markdoc.nodes.paragraph,
      attributes: {
        dir: {type: "string", default: "rtl"}
      }
    }
  }
};
```
